src/test/java/seedu/letsgethired/model/application/InternApplicationComparatorTest.java

```diff
@@ -3,51 +3,134 @@
+        InternApplicationComparator companyDescending = InternApplicationComparator
+                .getComparator(PREFIX_COMPANY, DESCENDING);
+
+        InternApplicationComparator cycleAscending = InternApplicationComparator
+                .getComparator(PREFIX_CYCLE, ASCENDING);
+        InternApplicationComparator cycleDescending = InternApplicationComparator
+                .getComparator(PREFIX_CYCLE, DESCENDING);
+
+        InternApplicationComparator roleAscending = InternApplicationComparator
+                .getComparator(PREFIX_ROLE, ASCENDING);
+        InternApplicationComparator roleDescending = InternApplicationComparator
+                .getComparator(PREFIX_ROLE, DESCENDING);
+
+        InternApplicationComparator statusAscending = InternApplicationComparator
+                .getComparator(PREFIX_STATUS, ASCENDING);
+        InternApplicationComparator statusDescending = InternApplicationComparator
+                .getComparator(PREFIX_STATUS, DESCENDING);
+
+        assertNotNull(companyAscending);
+        assertNotNull(companyDescending);
+
+        assertNotNull(cycleAscending);
+        assertNotNull(cycleDescending);
+
+        assertNotNull(roleAscending);
+        assertNotNull(roleDescending);
+
+        assertNotNull(statusAscending);
+        assertNotNull(statusDescending);
+
+        InternApplicationComparator deadlineDescending = InternApplicationComparator
+                .getComparator(PREFIX_DEADLINE, DESCENDING);
+
+        assertNull(deadlineDescending);
+    }
+
     @Test
     public void equals() {
-        InternApplicationComparator companyAscending = getComparator(PREFIX_COMPANY, ASCENDING);
-        InternApplicationComparator companyDescending = getComparator(PREFIX_COMPANY, DESCENDING);
-        InternApplicationComparator cycleAscending = getComparator(PREFIX_CYCLE, ASCENDING);
+        InternApplicationComparator companyAscending = InternApplicationComparator
+                .getComparator(PREFIX_COMPANY, ASCENDING);
+        InternApplicationComparator companyDescending = InternApplicationComparator
+                .getComparator(PREFIX_COMPANY, DESCENDING);
+
+        InternApplicationComparator cycleAscending = InternApplicationComparator
+                .getComparator(PREFIX_CYCLE, ASCENDING);
+
         assertNotNull(companyAscending);
         assertNotNull(companyDescending);
+
         assertNotNull(cycleAscending);
 
         // Same prefix and order -> returns true
-        assertEquals(companyAscending, getComparator(PREFIX_COMPANY, ASCENDING));
+        assertEquals(companyAscending, InternApplicationComparator.getComparator(PREFIX_COMPANY, ASCENDING));
 
         // Same prefix different order -> returns false
         assertNotEquals(companyAscending, companyDescending);
 
         // Different prefix same order -> returns false
         assertNotEquals(companyAscending, cycleAscending);
+    }
+
+    @Test
+    public void createCompositeComparator() {
```

Comment:
Is the composite comparator for those cases where `find` command is able to find for multiple different fields?